lib/gitlib.sh

```diff
@@ -299,8 +299,7 @@ gitlib::repo_state () {
     echo "${expectedRemote}" \
       | awk -F '[:/]' '{ print $(NF-1) "/" $NF }'
   )"
-  # make sure all '/' in the remote are escaped
-  remotePattern="${expectedRemote//\//\\\/}(.git)* \(fetch\)$"
+  remotePattern="${expectedRemote}(.git)* \\\(fetch\\\)$"
```

Comment:
I actually don't understand how it works with https in https://github.com/kubernetes/release/issues/730. So maybe I'm missing something here... :thinking: 

Without the extra backslashes, I get this locally:

```
[nikhita@nikhita-dev release]$ git remote -v | awk "-vP=${remotePattern}" '$0 ~ P {print $1}'
awk: warning: escape sequence `\(' treated as plain `('
awk: warning: escape sequence `\)' treated as plain `)'
``

Reply:
TL;DR: I think we should not use `awk` but `grep ... | cut ...` instead.

I did some tests [over there](https://gist.github.com/hoegaarden/d6d28ba3c6f4a89464b602ebf22baf10) (on `debian:sid` with `original-awk`, `gawk`, and `mawk` installed) and those seem to show:
- different `awk` implementations behave very different
- `grep -E ...` does not care if `/` are escaped or not

Therefore I opt for:
- `local expectedRemote="${RELEASE_TOOL_REPO:-[:/]kubernetes/release}"`
- `local remotePattern="${expectedRemote}(.git)* \(fetch\)$"`
- `local remote=$( git -C "$TOOL_ROOT" remote -v | grep -E "$remotePattern" | cut -f1 )`

Also, as side notes:
- https://github.com/kubernetes/release/blob/124d66d8967f7149c72984448987424ff60dbf41/lib/gitlib.sh#L298-L301 can be removed, as `remotePattern` will be set unconditionally on the next line
- I am confident that GNU grep and BSD grep behave differently too, but I don't care to support BSD grep here
- I am not sure, what we should do when multiple remotes match, but I don't care too much either as this whole thing should go away anyway soonish hopefully 🤞

Reply:
@idealhack @nikhita
🔥 🐚 🔥

Reply:
@hoegaarden thanks, updated :fire: